### PR TITLE
Split the web socket callback interface into two.

### PR DESCRIPTION
**Pull request by JakeWharton**
This separates the connection callbacks from the message callbacks. The goal here is making the `WebSocket` type standalone to be passed around before setting up the message callbacks. The new API on WebSocket, start(), must be called within the client's connect timeout for this to work.

This is a potential solution for https://github.com/square/retrofit/issues/924#issuecomment-115956713

### CODE REVIEW

File: okhttp-ws/src/main/java/com/squareup/okhttp/ws/WebSocketCallback.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ */
+package com.squareup.okhttp.ws;
+
+import com.squareup.okhttp.Response;
+import java.io.IOException;
+
+public interface WebSocketCallback {
+  /**
+   * Called when the request has successfully been upgraded to a web socket. You <b>must</b> call
+   * {@link WebSocket#start(WebSocket.Listener) start()} on the {@code WebSocket} with a listener.
```
Comment (swankjesse) [inline okhttp-ws/src/main/java/com/squareup/okhttp/ws/WebSocketCallback.java]: This interface feels labor intensive. Let's do a Hangout & you can better explain to me what the moving parts are!

Comment (JakeWharton) [inline okhttp-ws/src/main/java/com/squareup/okhttp/ws/WebSocketCallback.java]: :+1:

**Comment by swankjesse** [comment]
Is this ready for another look?

**Comment by JakeWharton** [comment]
No changes needed in OkHttp.

**Pull request closed without merging**